src/transform/class/PotentialClass.js

```diff
@@ -93,6 +100,38 @@ class PotentialClass {
   }
 
   createConstructor() {
-    return this.constructor.isEmpty() ? undefined : this.constructor.toMethodDefinition();
+    if (this.constructor.isEmpty()) {
+      return undefined;
+    }
+    else {
+      this.modifySuperCalls();
+      return this.constructor.toMethodDefinition();
+    }
+  }
+
+  modifySuperCalls() {
+    this.constructor.methodNode.body.body.forEach(body => {
```

Comment:
Worked on encapsulation in: 8200e98e04853053b5dfb6dd9edcbf2d10e05c74
Still have to work on where super() is called. It must be called first in order to be valid.